### PR TITLE
refactor(docfx): split page-related functions

### PR DESCRIPTION
**Pull request by coryan**
I plan to add additional functions related to pages, and the one source file was getting too big already.

Part of the work for #10895

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10963)

### CODE REVIEW

File: docfx/doxygen_pages.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_DOCFX_DOXYGEN_PAGES_H
+#define GOOGLE_CLOUD_CPP_DOCFX_DOXYGEN_PAGES_H
+
+#include <pugixml.hpp>
+#include <iosfwd>
```
Comment (dbolduc) [inline docfx/doxygen_pages.h:19]: needed in header?

File: docfx/doxygen_pages.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#include <string>
+#include <vector>
```
Comment (dbolduc) [inline docfx/doxygen_pages.h:21]: needed in header?

**Review by dbolduc — APPROVED** [review]

File: docfx/doxygen_pages.cc [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+std::string Page2Markdown(pugi::xml_node const& node) {
+  if (std::string_view{node.name()} != "compounddef" ||
+      std::string_view{node.attribute("kind").as_string()} != "page") {
+    std::ostringstream os;
+    os << "The node is not a page " << __func__ << "(): node=";
+    node.print(os, /*indent=*/"", /*flags=*/pugi::format_raw,
+               /*encoding=*/pugi::encoding_auto, /*depth=*/1);
+    throw std::runtime_error(std::move(os).str());
+  }
+  std::stringstream os;
```
Comment (devbww) [inline docfx/doxygen_pages.cc:87]: `ostringstream`?

File: docfx/doxygen_pages.cc [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "docfx/doxygen_pages.h"
+#include "docfx/doxygen2markdown.h"
+#include <iostream>
+#include <sstream>
+#include <string_view>
+#include <unordered_set>
```
Comment (devbww) [inline docfx/doxygen_pages.cc:20]: Unused?

File: docfx/doxygen_pages.h (hunk 1 above)
Comment (devbww) [inline docfx/doxygen_pages.h:19]: Unused?

File: docfx/doxygen_pages.h (hunk 2 above)
Comment (devbww) [inline docfx/doxygen_pages.h:21]: Unused?

Comment (coryan) [reply docfx/doxygen_pages.h]: Removed

File: docfx/doxygen_pages.h (hunk 1 above)
Comment (coryan) [reply docfx/doxygen_pages.h]: Removed

File: docfx/doxygen_pages.cc (hunk 4 above)
Comment (coryan) [reply docfx/doxygen_pages.cc]: Removed

File: docfx/doxygen_pages.cc (hunk 3 above)
Comment (coryan) [reply docfx/doxygen_pages.cc]: Fixed

File: docfx/doxygen_pages.h (hunk 2 above)
Comment (coryan) [reply docfx/doxygen_pages.h]: No, removed

File: docfx/doxygen_pages.h (hunk 1 above)
Comment (coryan) [reply docfx/doxygen_pages.h]: No, removed

File: docfx/doxygen_pages.cc [hunk 5]
```diff
@@ -17,7 +17,6 @@
 #include <iostream>
```
Comment (devbww) [inline docfx/doxygen_pages.cc:17]: Late breaking news: this looks unnecessary here (and in doxygen2markdown.cc).

**Review by devbww — APPROVED** [review]

File: docfx/doxygen_pages.cc (hunk 5 above)
Comment (coryan) [reply docfx/doxygen_pages.cc:17]: Fixed in both places, and found a missing `#include` too.

**Pull request merged**